### DISCUSSION TITLE
Please help me I'm stuck in setup foundry part plz 🙏

### DISCUSSION THREAD

**Discussion by audit-bykhizar** [Cyfrin/foundry-full-course-cu#2885] [Q&A] [answered]
I'm a Windows user, so I don't understand the part about setting up Foundry at all. Which terminal should I use to install Foundry, and what is the role of WSL? Where should 'Ubuntu' be used? Can someone please guide me? I'm not understanding this
![image_2024-11-19_225945650](https://github.com/user-attachments/assets/b84543b6-005b-4e3f-9875-82c51cb0911a)
![image_2024-11-19_225959619](https://github.com/user-attachments/assets/9472bd96-1004-4913-b374-9c3a173c67ff)
![image_2024-11-19_230019062](https://github.com/user-attachments/assets/4b812998-8046-4507-bfa8-5705ac22eb82)

**Comment by EngrPips** [comment]
I have been using Mac my whole life so I may not come in very handy here but check out this post https://medium.com/@regodefies/foundry-installation-on-windows-10-309407e39dee maybe it will help you.

**Reply by audit-bykhizar** [reply]
thanks bro for reply i will check it So if any one know this problem so let me know also

**Comment by 0xivanalexander** [comment, marked as answer]
I see your problem there my man, I m a mac user too but I recently needed to use a windows machine, I will try to explain as best as I can. 

To use Foundry on Windows, you will need to install WSL and a Linux distribution like Ubuntu. By

 running` wsl --install` in PowerShell as Administrator you will install wls. 
 
 Once set up, open Ubuntu to create a Linux environment where you can use shells like Bash or Zsh (I use zsh currently, but I see that Patrick uses bash, it doesnt really matter both are really similar"  . Inside Ubuntu, install Foundry with `curl -L https://foundry.paradigm.xyz` | bash followed by `foundryup`. To integrate this setup with VS Code, install the Remote - WSL extension, open a WSL session in VS Code, and use the integrated terminal to run commands.
 
  Ubuntu acts as your Linux workspace for tools like Foundry, accessible directly or through VS Code.
  
  Every time you enter vsCode from now on you must always switch to the linux terminal using the wsl command.
  
  I hope this helps!!
